addons/website_sale/models/account_move.py

```diff
@@ -21,6 +21,11 @@ def _auto_init(self):
             create_column(self.env.cr, "account_move", "website_id", "int4")
         super()._auto_init()
 
+    def get_base_url(self):
+        if len(self) == 1 and (website := self.website_id or self.partner_id.website_id):
+            return website.get_base_url()
+        return super().get_base_url()
```

Comment:
`get_base_url` can only be called on single records anyway, so the first part of the check seems unnecessary.

Also, I wouldn't trust nor use the partner `website_id` field.
Using the order or the move website seems enough (and safe).